packages/utils/telemetry-utils/src/test/sampledTelemetryHelper.spec.ts

```diff
@@ -233,6 +245,72 @@ describe("SampledTelemetryHelper", () => {
 		helper.dispose();
 		assert.strictEqual(logger.events.length, 1);
 	});
+
+	it("Correctly returns computed duration for custom data", () => {
+		const sampling = 10;
+		const helper = new SampledTelemetryHelper(
+			{ eventName: "testEvent" },
+			logger,
+			sampling,
+			true /* includeAggregateMetrics */,
+		);
+
+		const clock = sinon.useFakeTimers();
+		const startingPoint = 50; // Arbitrary starting point.
+		let totalDuration = 0;
+		let maxDuration = Number.MIN_VALUE;
+		let minDuration = Number.MAX_VALUE;
+
+		for (let i = 0; i < sampling; i++) {
+			helper.measure(() => {
+				const currentIterationDuration = startingPoint + i;
+
+				clock.tick(currentIterationDuration);
+				totalDuration += currentIterationDuration;
+				maxDuration = Math.max(maxDuration, currentIterationDuration);
+				minDuration = Math.min(minDuration, currentIterationDuration);
+			});
+		}
+
+		clock.restore();
+
+		assert.strictEqual(logger.events.length, 1);
+		assert.strictEqual(logger.events[0].totalDuration, totalDuration);
+		assert.strictEqual(logger.events[0].averageDuration, totalDuration / sampling);
+		assert.strictEqual(logger.events[0].maxDuration, maxDuration);
+		assert.strictEqual(logger.events[0].minDuration, minDuration);
+	});
+
+	it("Correctly returns computed averages and maxes for custom data", () => {
+		const sampling = 10;
+
+		const helper = new SampledTelemetryHelper<TestTelemetryProperties>(
+			{ eventName: "testEvent" },
+			logger,
+			sampling,
+		);
+
+		for (let i = 0; i < sampling; i++) {
+			helper.measure(() => {
+				return {
+					customData: {
+						propertyOne: i + 1,
+						propertyTwo: i + 2,
+						propertyThree: i + 3,
+						propertyFour: i + 4,
```

Comment:
The biggest concern of this PR. Still not correctly throwing errors when additional properties are added to the `customData` object.